google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1319,9 +1319,9 @@ def extract extract_url, format: nil, compression: nil, delimiter: nil,
         # file directly. See [Loading Data with a POST Request](
         # https://cloud.google.com/bigquery/loading-data-post-request#multipart).
         #
-        # @param [File, Google::Cloud::Storage::File, String] file A file or the
-        #   URI of a Google Cloud Storage file containing data to load into the
-        #   table.
+        # @param [File, Google::Cloud::Storage::File, String, URI] file A file
```

Comment:
To explicitly support an URI object, I would add a conditional to the beginning of the method that checks if the object is a URI and convert it to a String. Or, try to parse any string input with URI and handle the argument as a URI internally.